### PR TITLE
Correct gene search warning

### PR DESCRIPTION
**Pull request by northwestwitch**
Partly fixes this: https://github.com/Clinical-Genomics/scout/issues/2614#issuecomment-842884213

Basically warn when a gene symbol searched in variant filters is found in clinical panel but the panel has an outdated symbol. Current situation is that you get a warning that gene is not found in clinical panel.

**How to test --> locally**:
- [x] Test situation when you search a gene by HGNC ID and you find it: for instance gene 17284 (should return gene list with POT1)
- [x] Test situation when you search a gene by symbol and you find it: for instance gene POT1 (should return gene list with POT1)
- [x] Test situation when you search a gene by HGNC ID and you DON'T find it: for instance gene 666 (should flash an alert that HGNC id not found)
- [x]  Test situation when you search a gene by symbol and you DON'T find it: for instance gene POTTA (should flash an alert that HGNC symbol not found)
- Modify panel1 in database (gene n. 99: POT1) and replace gene symbol with POTTA
- [x] Situation when you search a gene by symbol (POT1) and panel symbol is outdated (should flash an alert that clinical list contains an outdated symbol for POT1)

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: scout/server/blueprints/variants/controllers.py [hunk 1]
```diff
@@ -1008,31 +1067,10 @@ def update_form_hgnc_symbols(store, case_obj, form):
 
     # check if supplied gene symbols exist and are clinical
     is_clinical = form.data.get("variant_type", "clinical") == "clinical"
-    clinical_symbols = store.clinical_symbols(case_obj) if is_clinical else None
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/controllers.py:1011]: Moving this part to another function

**Comment by northwestwitch** [comment]
I'll add a test

File: scout/server/blueprints/variants/controllers.py [hunk 2]
```diff
@@ -963,6 +963,67 @@ def populate_sv_filters_form(store, institute_obj, case_obj, category, request_o
...
+
+    for hgnc_symbol in hgnc_symbols:
+        hgnc_gene = store.hgnc_genes_find_one(hgnc_symbol, genome_build)
+
+        if hgnc_gene is None:
+            not_found_symbols.append(hgnc_symbol)
+        elif is_clinical and hgnc_gene["hgnc_id"] in clinical_hgnc_ids:
+            updated_hgnc_symbols.append(hgnc_symbol)
+            if hgnc_symbol not in clinical_symbols:
+                # clinical symbols from gene panels might not be up to date with latest gene names
```
[the remark this answers is not in the document]
Comment (dnil) [reply scout/server/blueprints/variants/controllers.py:1000]: Which would then be caught first. 😊 Ok, good!

**Review by dnil — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -30,8 +30,9 @@ About changelog [here](https://keepachangelog.com/en/1.0.0/)
 - Causative view sort out of memory error
 - Use hgnc_id for gene filter query
 - Typo in case controllers displaying an error every time a patient is matched against external MatchMaker nodes
-- Do not crash while attemping an update for variant documents that are too big (> 16 MB)
+- Do not crash while attempting an update for variant documents that are too big (> 16 MB)
 - Old STR causatives may not have HGNC symbols - fix sort lambda
+- Warn if gene is a present in clinical panel with an outdated name when filtering variants by gene symbol
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply CHANGELOG.md:35]: Hahaha yes!

File: scout/server/blueprints/variants/controllers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/blueprints/variants/controllers.py:1000]: Yes the gene is not found in that case

File: scout/server/blueprints/variants/controllers.py [hunk 4]
```diff
@@ -977,17 +1038,15 @@ def update_form_hgnc_symbols(store, case_obj, form):
     """
 
     hgnc_symbols = []
-    non_clinical_symbols = []
-    not_found_symbols = []
     not_found_ids = []
-    updated_hgnc_symbols = []
+    genome_build = "38" if str(case_obj.get("genome_build")) == "38" else "37"
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/blueprints/variants/controllers.py:1042]: Yes better your way!

**Comment by northwestwitch** [comment]
I'm adding another test for the controllers function and then I think I'm good to merge!

File: scout/server/blueprints/variants/controllers.py [hunk 5]
```diff
@@ -963,6 +963,67 @@ def populate_sv_filters_form(store, institute_obj, case_obj, category, request_o
...
+        else:
+            non_clinical_symbols.append(hgnc_symbol)
+
+    errors = {
+        "non_clinical_symbols": {
+            "alert": "Gene not included in clinical list",
+            "gene_list": non_clinical_symbols,
+        },
+        "not_found_symbols": {"alert": "HGNC symbol not found", "gene_list": not_found_symbols},
+        "not_found_ids": {"alert": "HGNC id not found", "gene_list": not_found_ids},
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/blueprints/variants/controllers.py:1012]: It is passed from the previous function, isn't it?

**Comment by northwestwitch** [comment]
Why is this test failing now (tests/server/blueprints/variants/test_variants_views.py::test_variants_clinical_filter), I haven't even touched that file!

File: scout/server/blueprints/variants/controllers.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (dnil) [reply scout/server/blueprints/variants/controllers.py:1012]: Right, right!

**Comment by northwestwitch** [comment]
I got stuck with this. I want to understand why that test is failing now

**Comment by dnil** [comment]
> I got stuck with this. I want to understand why that test is failing now

Absolutely - it is at least in a view touched. I'll get there soon as well! Some coffee first..

**Comment by northwestwitch** [comment]
Ok it was actually those 2 tests. Perhaps I'm modifying something in the tests that is later used in the variants.views?

**Comment by northwestwitch** [comment]
YES!!! tests work now, can I merge @dnil ??

**Pull request merged**